File: 401-500/438.Find All Anagrams in a String.py
class Solution(object):
	def findAnagrams(self, s, p):
		lenp=len(p)
		lens=len(s)
		result=[]
		for i in range(0,lens-lenp):
			test=s[i:i+lenp]
			l1=list(p)
			l2=list(test)
			l1.sort()
			l2.sort()
			if l1==l2:
				result.append(i)
		return result

File: 401-500/438.Find All Anagrams in a String.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
        vector<int> pv(256,0), sv(256,0), res;
        if(s.size() < p.size())
           return res;
        for(int i = 0; i < p.size(); ++i)
        {
            ++pv[p[i]];
            ++sv[s[i]];
        }
        if(pv == sv)
           res.push_back(0);
        for(int i = p.size(); i < s.size(); ++i)
        {
            ++sv[s[i]];
            --sv[s[i-p.size()]];
            if(pv == sv)
               res.push_back(i-p.size()+1);
        }
        return res;
    }
};
//网上答案，python版本可运行但超时